### PR TITLE
WIP: More automated tests

### PR DESCRIPTION
**Pull request by callebtc**

### CODE REVIEW

File: lnbits/core/views/api.py [hunk 1]
```diff
@@ -359,7 +366,7 @@ async def api_payments_sse(
     request: Request, wallet: WalletTypeInfo = Depends(get_key_type)
 ):
     return EventSourceResponse(
-        subscribe(request, wallet), ping=20, media_type="text/event-stream"
+        subscribe(request, wallet.wallet), ping=20, media_type="text/event-stream"
```
Comment (callebtc) [inline lnbits/core/views/api.py:369]: This was probably an undetected bug.

File: lnbits/core/views/api.py [hunk 2]
```diff
@@ -513,15 +527,19 @@ async def api_lnurlscan(code: str):
...
 
+class DecodePayment(BaseModel):
+    data: str
+
+
 @core_app.post("/api/v1/payments/decode")
-async def api_payments_decode(data: str = Query(None)):
-    print(data)
+async def api_payments_decode(data: DecodePayment):
+    payment_str = data.data
```
Comment (callebtc) [inline lnbits/core/views/api.py:536]: Rebased from PR #568

File: lnbits/core/views/api.py [hunk 3]
```diff
@@ -231,12 +236,9 @@ async def api_payments_pay_invoice(bolt11: str, wallet: Wallet):
     status_code=HTTPStatus.CREATED,
 )
 async def api_payments_create(
-    wallet: WalletTypeInfo = Depends(get_key_type),
+    wallet: WalletTypeInfo = Depends(require_invoice_key),
```
Comment (callebtc) [inline lnbits/core/views/api.py:239]: Rebased from PR #565

File: lnbits/core/views/api.py [hunk 4]
```diff
@@ -145,6 +146,7 @@ async def api_payments_create_invoice(data: CreateInvoiceData, wallet: Wallet):
     if data.unit == "sat":
         amount = int(data.amount)
     else:
+        assert data.unit is not None, "unit not set"
```
Comment (callebtc) [inline lnbits/core/views/api.py:149]: for mypy

File: lnbits/core/views/api.py [hunk 5]
```diff
@@ -305,7 +312,7 @@ async def api_payments_pay_lnurl(
         extra["success_action"] = params["successAction"]
     if data.comment:
         extra["comment"] = data.comment
-
+    assert data.description is not None, "description is required"
```
Comment (callebtc) [inline lnbits/core/views/api.py:315]: For mypy

File: lnbits/core/views/api.py (hunk 1 above)
Comment (arcbtc) [reply lnbits/core/views/api.py:369]: This explains a lot!

**Pull request merged**